client/src/js/components/Bus.js

```diff
@@ -281,90 +330,153 @@ const createStatusComponent = (status, t) => {
-              <FontAwesomeIcon icon={faAngleDoubleRight} size="1x" />{" "}
-              {t("bus.Waseda")}
-            </StyledSubHeading>
-            <Status>{nishiStatusComponent.status}</Status>
-            <Remark>{nishiStatusComponent.remark}</Remark>
-          </BusStatus>
-          <ModalContainer
-            linkText={t("bus.Add to home screen")}
-            text={t("bus.and never miss a bus again!")}
-          >
-            <ModalArticle>
-              <ModalSection>
-                <ModalHeading>Android / Chrome:</ModalHeading>
-                <p>
-                  {t("bus.Tap on the top-right icon")}
-                  &nbsp;
-                  <FontAwesomeIcon icon={faEllipsisV} size="1x" />
-                  &nbsp;
-                  {t("bus.and select Add to Home screen")}
-                </p>
-                <ModalImage
-                  src={a2hsChrome}
-                  alt="Add to home screen image for Chrome"
+const DateSelector = ({ value, onClick }) => (
+  <DatePickerButton onClick={onClick}>
+    <FontAwesomeIcon icon={faCalendarAlt} size="1x" />{" "}
+    {value}
+  </DatePickerButton>
+);
+
+const TimeSelector = ({ value, onClick }) => (
+  <DatePickerButton onClick={onClick}>
+    <FontAwesomeIcon icon={faClock} size="1x" />{" "}
+    {value}
+  </DatePickerButton>
+);
+
+class Bus extends React.Component {
+  state = {
+    date: new Date()
+  };
+
+  onDatetimeChange = date => {
+    this.setState({ date: date || new Date() })
+  }
+
+  clearDatetime = () => {
+    this.setState({ date: new Date() })
+  }
+
+  render () {
+    const { t, lng } = this.props;
+    const { wasedaStatus, nishiStatus } = getBusStatuses(this.state.date, lng, t);
+    const wasedaStatusComponent = createStatusComponent(wasedaStatus, t);
+    const nishiStatusComponent = createStatusComponent(nishiStatus, t);
+    return (
+      <Wrapper>
+        <Helmet>
+          <title>WasedaTime -　Bus</title>
+          <meta
+            name="description"
+            content="Shuttle Bus Arrival Time Checking at Waseda University."
+            />
+          <meta property="og:title" content="WasedaTime - Bus" />
+          <meta
+            property="og:description"
+            content="Shuttle Bus Arrival Time Checking at Waseda University."
+            />
+          <meta property="og:site_name" content="WasedaTime - Bus" />
+        </Helmet>
+        <ExtendedOverlay>
+          <InfoWrapper>
+            <StyledHeading>{t("bus.busStatus")}</StyledHeading>
+            <p>
+              <FontAwesomeIcon icon={faSearch} size="1x" />{' '}
+              {t("bus.Assign a date / time to check the next bus")}：
+            </p>
+
+            <DatetimeSelection>
+              <DatePickerSpan>
+                <DatePicker
+                  selected={this.state.date}
+                  onChange={date => this.onDatetimeChange(date)}
+                  dateFormat="yyyy-MM-dd"
+                  customInput={<DateSelector />}
```

Comment:
I found an error caused by `DateSelector` (same for `TimeSelector`)
![Screen Shot 2020-06-04 at 15 22 58](https://user-images.githubusercontent.com/30998659/83731937-e4532b00-a67d-11ea-8dcf-34d8b32d6f38.png)
I think you can fix it using `forwardRef` (an example is available [here](https://github.com/atomiks/tippyjs-react#component-children))
Don't forget to check the errors in your browser console next time~

Reply:
I have added it. Please check!